Resources/views/Button/create_button.html.twig

```diff
@@ -11,20 +11,27 @@ file that was distributed with this source code.
 
 {% if admin.hasAccess('create') and admin.hasRoute('create') %}
     {% if admin.subClasses is empty %}
-        <li>
-            <a class="sonata-action-element" href="{{ admin.generateUrl('create') }}">
+        <a class="sonata-action-element" href="{{ admin.generateUrl('create') }}">
+            <i class="fa fa-plus-circle" aria-hidden="true"></i>
+            {{ 'link_action_create'|trans({}, 'SonataAdminBundle') }}</a>
+    {% else %}
+        {% if 1 is same as (admin.subclasses|length) %}
+            {% set subclass = admin.subclasses|keys|first %}
+            <a href="{{ admin.generateUrl('create', {'subclass': subclass}) }}">
                 <i class="fa fa-plus-circle" aria-hidden="true"></i>
-                {{ 'link_action_create'|trans({}, 'SonataAdminBundle') }}
+                {{ 'link_action_create'|trans({}, 'SonataAdminBundle') }} {{ subclass|trans({}, admin.translationdomain) }}
             </a>
-        </li>
-    {% else %}
-        {% for subclass in admin.subclasses|keys %}
-            <li>
-                <a class="sonata-action-element" href="{{ admin.generateUrl('create', {'subclass': subclass}) }}">
-                    <i class="fa fa-plus-circle" aria-hidden="true"></i>
-                    {{ 'link_action_create'|trans({}, 'SonataAdminBundle') }} {{ subclass|trans({}, admin.translationdomain) }}
-                </a>
-            </li>
-        {% endfor %}
+        {% else %}
+            <ul class="list-unstyled">
+                {% for subclass in admin.subclasses|keys %}
+                    <li>
+                        <a href="{{ admin.generateUrl('create', {'subclass': subclass}) }}">
+                            <i class="fa fa-plus-circle" aria-hidden="true"></i>
+                            {{ 'link_action_create'|trans({}, 'SonataAdminBundle') }} {{ subclass|trans({}, admin.translationdomain) }}
+                        </a>
+                    </li>
```

Comment:
Maybe create a new block just for the link and call it to avoid the duplication?